### PR TITLE
[voicerepublic] Add new extractor

### PR DESCRIPTION
**Pull request by duncankl**
Hello! Just a little extractor for Voice Republic, which hosts spoken word audio content.

### CODE REVIEW

File: youtube_dl/extractor/voicerepublic.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    _TEST = {
+        'url': 'https://voicerepublic.com/talks/watching-the-watchers-building-a-sousveillance-state',
+        'md5': '0554a24d1657915aa8e8f84e15dc9353',
+        'info_dict': {
+            'id': '2296',
+            'ext': 'm4a',
+            'title': 'Watching the Watchers: Building a Sousveillance State',
+            'thumbnail': 'https://voicerepublic.com/system/flyer/2296.png',
+            'description': 'md5:715ba964958afa2398df615809cfecb1',
+            'creator': 'M. C. McGrath',
```
Comment (dstftw) [inline youtube_dl/extractor/voicerepublic.py]: There is no such field, must be `uploader`.

File: youtube_dl/extractor/voicerepublic.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        req = compat_urllib_request.Request(url)
+        # Older versions of Firefox get redirected to an "upgrade browser" page
+        req.add_header('User-Agent', 'youtube-dl')
+        webpage = self._download_webpage(req, display_id)
+        thumbnail = self._og_search_thumbnail(webpage)
+        video_id = self._search_regex(r'/(\d+)\.png', thumbnail, 'id')
+
+        if '<div class=\'vr-player jp-jplayer\'' in webpage:
+            formats = [{
+                'url': 'https://voicerepublic.com/vrmedia/{}-clean.{}'.format(video_id, ext),
```
Comment (dstftw) [inline youtube_dl/extractor/voicerepublic.py]: Change to `{0}` and `{1}` .

File: youtube_dl/extractor/voicerepublic.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                'ext': ext,
+                'format_id': ext,
+                'vcodec': 'none',
+            } for ext in ['m4a', 'mp3', 'ogg']]
+            self._sort_formats(formats)
+        else:
+            # Audio is still queued for processing
```
Comment (dstftw) [inline youtube_dl/extractor/voicerepublic.py]: You may want to output this meaningful error with `ExtractorError` if this is for sure.

File: youtube_dl/extractor/voicerepublic.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            formats = []
+
+        return {
+            'id': video_id,
+            'title': self._og_search_title(webpage),
+            'formats': formats,
+            'url': self._og_search_url(webpage),
+            'thumbnail': thumbnail,
+            'description': self._og_search_description(webpage),
+            'creator': self._search_regex(r'<meta content=\'([^\']*?)\' name=\'author\'>', webpage, 'author', fatal=False),
```
Comment (dstftw) [inline youtube_dl/extractor/voicerepublic.py]: Use `self._html_search_meta` instead.

**Comment by duncankl** [comment]
Thanks for the feedback, I've added a series of commits to address the comments.

File: youtube_dl/extractor/voicerepublic.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        video_id = self._search_regex(r'/(\d+)\.png', thumbnail, 'id')
+
+        if '<a>Queued for processing, please stand by...</a>' in webpage:
+            raise ExtractorError('Audio is still queued for processing')
+
+        ext_matches = re.finditer(r'data-\w+=\'/vrmedia/\d+-clean\.(\w+)\'', webpage)
+        exts = [match.group(1) for match in ext_matches]
+
+        formats = [{
+            'url': 'https://voicerepublic.com/vrmedia/{}-clean.{}'.format(video_id, ext),
```
Comment (dstftw) [inline youtube_dl/extractor/voicerepublic.py]: You still hardcode the path schema. If it changes to, say, `/vrmedia2/<id>-clean.<ext>` the extraction is broken then. You need to capture the whole path:
```
for ext, path in re.findall(r"data-([^=]+)='([^']+)'", webpage):
    format_url = https://voicerepublic.com + path
    ...
```

**Comment by duncankl** [comment]
Ah yes, I see what you mean. This latest version no longer depends on a hardcoded path schema. I made the regex a bit more strict to prevent it matching other `data-` tags on the page.

**Comment by dstftw** [comment]
Thanks.

**Comment by duncankl** [comment]
Great! Thank you for your help @dstftw.

**Comment by dstftw** [comment]
Do you want to be credited as one of the authors of youtube-dl? If so under what name?

**Comment by duncankl** [comment]
Actually I'm already in the authors list; I've added one other extractor previously. Thanks for checking though.

**Pull request merged**